### ISSUE TITLE
Create admin ability to pause locks/burns on Sifnode for Peggy 2.0 deployment

### ISSUE THREAD

**Issue by pandaring2you** [Sifchain/sifnode#2445]
Ability to pause locks/burns on Sifnode so that we can deploy peggy 2.0 changes. This functionality won't be needed until mid March.

**Comment by Jedi2002** [comment]
To get some clarity on requirements is the following accurate 

# Requirements
- Add a new transaction to pause lock tx and burn tx .
    - Sender can only be admin
- Add new transaction to unpause lock and burn tx
    - Sender can only be admin

**Comment by pandaring2you** [comment]
yup - basically, we want to stop any imports and exports from happening so that we can deploy the peggy 2.0 changes. if there are any other types of transactions that might affect sifnode while we're deploying, please call that out as well. this is all we could think of when meeting with tim

**Comment by Jedi2002** [comment]
Alright that helps, 
When we say pause does that mean 
- Reject incoming Lock/Burn transactions ( Easier to implement ) 
- Add the transactions to a queue, which will be processed once the transactions are unpaused (Much more difficult to implement)

**Comment by Brando753** [comment]
To clarify this ask, pauser functionality already exists in the smart contracts. It currently allows a pauser to pause or unpause this system. We would like you to spilt out the roles so that there are two RBAC roles one for pausing and one for unpausing. The theory is that there should be many people allowed to pause the bridge (trusted developers, relayers, witnesses, etc) but only a few could unpause it.

**Comment by pandaring2you** [comment]
@Brando753 - wait are you sure? i thought separating the pauser/unpauser roles on our smart contracts was JP's task: https://app.zenhub.com/workspaces/current-sprint---engineering-615a2e9fe2abd5001befc7f9/issues/sifchain/sifnode/1910 

This task we talked about with Tim about pausing imports/exports on the sifnode side.

**Comment by timlind** [comment]
@Jedi2002 it should be a simple reject system when paused. 

In terms of roles I think we'll just have one role that can pause and unpause, we could make them different addresses if need be but is that really necessary?

**Comment by pandaring2you** [comment]
Would need a different admin role in a separate module. 
Est 1 week

**Comment by Jedi2002** [comment]
Hey @marshsif , I can start on this task if required , 


# Requirements
Add a new transaction to pause lock tx and burn tx .
Add new transaction to unpause lock and burn tx

- Reject all lock and burn transactions when paused .
- Admin control over these transactions is required .

**Comment by marshsif** [comment]
OK @Jedi2002 , please do, thanks!

**Comment by Jedi2002** [comment]
This pr adds the pause functionality 
https://github.com/Sifchain/sifnode/pull/2665

**Comment by Jedi2002** [comment]
Hey @pandaring2you Who from peggy team should I tag for pr reviews? The sifnode team is busy with pmtp work rn , it might make more sense for someone from preggy to be reviewing the pr's

**Comment by pandaring2you** [comment]
perhaps @juniuszhou cc @Brando753

**Comment by pandaring2you** [comment]
@smartyalgo is currently reviewing

**Comment by pandaring2you** [comment]
@Jedi2002 - Do you know how much more time it would take to address Michael's comments on the PR and close this out? Would be great to close this out after your work on .45 rather than hand this off to another dev

**Comment by pandaring2you** [comment]
@Jedi2002 syncing with @smartyalgo today to review PR. Likely not an issue

**Comment by pandaring2you** [comment]
From @timlind regarding #2665 : "Thanks. I think we need to get this scheduled for a release before we can merge though."
Hey @timlind - can you explain? Why can't we merge?

**Comment by timlind** [comment]
@pandaring2you we are in the process of deprecating develop branch and don't want work there that isn't ready to ship. Can this be merged into a feature branch instead?

**Comment by pandaring2you** [comment]
@Brando753 - do you have a recommendation here? can we merge this with future/peggy2?

**Comment by pandaring2you** [comment]
@Jedi2002 - if you have time outside of margin - can you review your PR again and see if we can merge this functionality into trunk? cc @smartyalgo @Brando753

**Comment by Jedi2002** [comment]
Hey @pandaring2you this work was done a long time ago, Not sure why it was never merged, I do see @smartyalgo approved it . I do not have access to merge to develop. 
This needs to be redone now, as the sifnoide code has changed a lot since then, updating the PR and redoing it might almost take the same time .
We should be wrapping up margin pretty soon , at least the V1 , please check with @brianosaurus and assign it to me , I can pick this up after .

**Comment by pandaring2you** [comment]
@smartyalgo to take this on

**Comment by smartyalgo** [comment]
This should be fine once we rebase the PR onto master. Reached out to him, will see if anything comes up

**Comment by pandaring2you** [comment]
Will get this in tonight for deployment on monday

**Comment by pandaring2you** [comment]
Aiming to have integration test done by EOD tomorrow

**Comment by pandaring2you** [comment]
Will get into v1.0.15beta

**Comment by pandaring2you** [comment]
Testing should be finished by tomorrow. cc @ghuznee

**Comment by pandaring2you** [comment]
Local testing completed and merged to master. Deployed to testnet. Need to run tests on testnet. Working with Lance and Caner on this for next release

**Comment by pandaring2you** [comment]
@smartyalgo : completed testnet testing, supporting Caner with regression testing. No pending need from him

@smartyalgo - so is this fully tested and ready for release for .15?
